src/ir/possible-contents.cpp

```diff
@@ -896,6 +896,26 @@ struct InfoCollector
     }
     addRoot(curr, PossibleContents::exactType(curr->type));
   }
+  void visitArrayNewSeg(ArrayNewSeg* curr) {
+    if (curr->type == Type::unreachable) {
+      return;
+    }
+    auto heapType = curr->type.getHeapType();
+    switch (curr->op) {
+      case NewData: {
+        Type elemType = heapType.getArray().element.type;
+        addRoot(DataLocation{heapType, 0},
+                PossibleContents::fromType(elemType));
+        return;
+      }
+      case NewElem: {
+        Type segType = getModule()->elementSegments[curr->segment]->type;
+        addRoot(DataLocation{heapType, 0}, PossibleContents::fromType(segType));
```

Comment:
Sorry for missing this in review, this should be
```
info.links.push_back({DataLocation{heapType, 0}, PossibleContents::fromType(segType)});
addRoot(curr, PossibleContents::exactType(curr->type));
```

Reply:
Actually the first line should be a root as well, sorry, since we know nothing about it but the type.